AutoGear.lua

```diff
@@ -1889,80 +1891,99 @@ AutoGearFrame:SetScript("OnEvent", function (this, event, arg1, arg2, arg3, arg4
 		elseif (event == "QUEST_DETAIL") then
 			QuestDetailAcceptButton_OnClick()
 		elseif (event == "GOSSIP_SHOW") then
-			--active quests
-			if (not IsClassic) then
-				local quests = C_GossipInfo.GetNumActiveQuests()
-				local info = {C_GossipInfo.GetActiveQuests()}
-				for i = 1, quests do
-					local title, questLevel, isTrivial, frequency, repeatable, isComplete, isLegendary = info[(i-1)*6+1], info[(i-1)*6+2], info[(i-1)*6+3], info[(i-1)*6+4], info[(i-1)*6+5], info[(i-1)*6+6], info[(i-1)*6+7]
-					if (isComplete) then
+			--active quests			
+			if (SL) then
+				for i = 1, C_GossipInfo.GetNumActiveQuests() do
+					local quest = C_GossipInfo.GetActiveQuests()[i]
+					if (quest["isComplete"]==true) then
 						C_GossipInfo.SelectActiveQuest(i)
 					end
 				end
-			else
-				local quests = GetNumGossipActiveQuests()
+			end
+			if (IsClassic or BFA) then
 				local info = {GetGossipActiveQuests()}
-				for i = 1, quests do
+				for i = 1, GetNumGossipActiveQuests() do
 					local name, level, isTrivial, isComplete, isLegendary = info[(i-1)*6+1], info[(i-1)*6+2], info[(i-1)*6+3], info[(i-1)*6+4], info[(i-1)*6+5]
 					if (isComplete) then
 						SelectGossipActiveQuest(i)
 					end
 				end
-			end
-			--available quests
-			if (not IsClassic) then
-				quests = C_GossipInfo.GetNumAvailableQuests()
-				info = {C_GossipInfo.GetAvailableQuests()}
-				for i = 1, quests do
-					local title, questLevel, isTrivial, frequency, repeatable, isComplete, isLegendary = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5], info[(i-1)*7+6], info[(i-1)*7+7]
-					if (not isTrivial) then
-						C_GossipInfo.SelectAvailableQuest(i)
+			end			
+			--available quests			
+			if (SL) then 
+				for i = 1, C_GossipInfo.GetNumAvailableQuests() do
+					local quest = C_GossipInfo.GetAvailableQuests()[i]
+					if (quest["isTrivial"]==false) then
+	  					C_GossipInfo.SelectAvailableQuest(i)
 					end
 				end
-			else
-				quests = GetNumGossipAvailableQuests()
+			end
+			if (BFA or IsClassic) then
 				info = {GetGossipAvailableQuests()}
-				for i = 1, quests do
-				local name, level, isTrivial, frequency, isRepeatable = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5]
+				for i = 1, GetNumGossipAvailableQuests() do
+					local name, level, isTrivial, frequency, isRepeatable = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5]
 					if (not isTrivial) then
 						SelectGossipAvailableQuest(i)
 					end
 				end
 			end
 		elseif (event == "QUEST_GREETING") then
 			--active quests
-			if (not IsClassic) then
-				local quests = C_GossipInfo.GetNumActiveQuests()
-				for i = 1, quests do
+			if (SL) then
```

Comment:
Please rename to ```IsSL``` to follow the convention and check for ```IsBFA``` here instead of the third section here, as that section is for Classic only.